news/2019-02-07-weld-310Final.asciidoc

```diff
@@ -0,0 +1,81 @@
+** Added new SPI for class defining, deprecated old approach in `ProxyServices` (link:https://issues.jboss.org/browse/WELD-2556[WELD-2556, window="_blank"])
+*** Integrators will have to implement this new API to be able to operate on JDK 12+
+*** Weld now delegates class defining to integrators in order to avoid having to crack open `ClassLoader` methods
+*** Previously, integrators provided `ClassLoader` instance which Weld then used to invoke `defineClass()` methods
+*** We not ask integrators to invoke those methods themselves while providing them with all the necessary bits for doing so
+** Remove deprecated parts of API/SPI (link:https://issues.jboss.org/browse/WELD-2558[WELD-2558, window="_blank"])
+*** Lots of unused and long deprecated methods were removed, should have no effect on code as those weren't used anyway (most of them since Weld 3.0.0)
+
+* Weld Core
+** Weld will now log a `WARNING` if you try to register an invalid qualifier (link:https://issues.jboss.org/browse/WELD-2522[WELD-2522, window="_blank"])
+** You can now use `InterceptionFactory` with an interface as a parameter (link:https://issues.jboss.org/browse/WELD-2550[WELD-2550, window="_blank"], link:https://issues.jboss.org/browse/WELD-2533[WELD-2533, window="_blank"])
+*** This means the proxy class will be based off an interface which is by definition always proxyable
+*** You can therefore even supply an unproxyable implementation and it will still work
+*** Note that this is experimental feature with some limitations to it, see link:http://docs.jboss.org/weld/reference/latest/en-US/html_single/#_loosening_the_limitations_of_literal_interceptionfactory_literal[Weld docs] for more details
+** Correct atomic behaviour in `RequestContextController` (link:https://issues.jboss.org/browse/WELD-2536[WELD-2536, window="_blank"])
+** Fix rare race condition in `ConcurrentValidator` (link:https://issues.jboss.org/browse/WELD-2545[WELD-2545, window="_blank"])
+** Small correction to interceptor resulution when they have no bindings (link:https://issues.jboss.org/browse/WELD-2521[WELD-2521, window="_blank"])
+** Enforce consistent behaviour between `AnnotatedType` and `WithAnnotations` in regards to how default methods (link:https://issues.jboss.org/browse/WELD-2551[WELD-2551, window="_blank"])
+** All proxy-specific methods added by Weld now have more complex names to avoid (very rare) method clashes (link:https://issues.jboss.org/browse/WELD-2508[WELD-2508, window="_blank"])
+** Lower logging level of `InterceptorLogger.unableToDetermineInterceptedBean()` from `WARN` to `INFO` (link:https://issues.jboss.org/browse/WELD-2546[WELD-2546, window="_blank"])
+
+* Weld SE
+** You can now extend the set of bean defining annotations in Weld SE (link:https://issues.jboss.org/browse/WELD-2523[WELD-2523, window="_blank"])
+*** This is handy for when you have discovery on and want custom beans automatically picked up while still avoiding the `all` discovery mode
```

Comment:
This is handy for when you have.. - is the "for" needed here?

Reply:
Optional, should work both ways.